### ISSUE TITLE
keter-1.4.0 hangs up

### ISSUE THREAD

**Issue by builtinnya** [snoyberg/keter#96]
I tried keter-1.4.0 on my VPS running Ubuntu but got "Out of memory" error after about 30 seconds.
It is similar to #13.

### Command
```bash
$ sudo /opt/keter/bin/keter /opt/keter/etc/keter-config.yaml
```

### Running Environment
OS:
```bash
$ lsb_release -a
No LSB modules are available. 
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.2 LTS 
Release:        14.04
Codename:       trusty
```

GHC:
```bash
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.6.3
```

Cabal:
```bash
$ cabal --version
cabal-install version 1.16.0.2
using version 1.16.0 of the Cabal library
```

Keter:
```bash
$ /opt/keter/bin/keter --version
keter version: 1.4.0
```

### Configuration
 /opt/keter/etc/keter-config.yaml:
```yaml
root: ..
listeners:
    - host: "*4"
```

### Tentative Solution
I switched to keter-1.3.10.1:
```bash
$ cabal install keter-1.3.10.1
```
and now everything is working fine.
So something seems wrong with keter-1.4.0.

**Comment by snoyberg** [comment]
Thanks for the very solid report, this was a bad bug on my part. Should be fixed in 1.4.0.1, can you confirm?

**Comment by builtinnya** [comment]
I confirmed keter-1.4.0.1 worked correctly on my environment.
Thank you for the quick fix.
